Source/Mdk.CommandLine/IngameScript/Pack/DefaultProcessors/PreprocessorConditionals.cs

```diff
@@ -33,6 +33,18 @@ public async Task<Document> ProcessAsync(Document document, IPackContext context
         Stack<Block> stack = new();
         stack.Push(root);
         var needsMacroCheck = false;
+
+
+        ImmutableHashSet<string>.Builder allExistingLocalDefineBuilder = ImmutableHashSet.CreateBuilder<string>();
```

Comment:
You just need the allSymbols list. You probably use a HashSet<string> in place of a list for quick duplicate rejection. Either way there's no need for multiple lists here, everything must end up in the same place anyway.